test/remote-syslog.sh

```diff
@@ -9,189 +9,301 @@ source "${OS_O_A_L_DIR}/hack/testing/util.sh"
 os::util::environment::use_sudo
 
 FLUENTD_WAIT_TIME=${FLUENTD_WAIT_TIME:-$(( 2 * minute ))}
+MUX_WAIT_TIME=$(( 5 * minute ))
+ALTPORT=601
 
 os::test::junit::declare_suite_start "Remote Syslog Configuration Tests"
 
+# switch pods type depending on the mux configuration
+fluentdtype="fluentd"
+if oc get dc/logging-mux > /dev/null 2>&1 ; then
+    os::log::debug "$( oc get dc/logging-mux )"
+    fluentdtype="mux"
+fi
+
 # save daemonset
 saveds=$( mktemp )
 oc export ds/logging-fluentd -o yaml > $saveds
 
 # restore configs back to how it was before we ran our tests
 function reset_fluentd_daemonset() {
-  os::log::info Restoring original fluentd daemonset environment variable
+  os::log::info Restoring original fluentd daemonset / mux deploymentconfig environment variable
   os::log::debug "$( oc replace -f $saveds )"
 }
 
-# bz1515715 -- check if mux is enabled.
-savemuxdc=""
-if oc get dc/logging-mux > /dev/null 2>&1 ; then
-  savemuxdc=$( mktemp )
-  oc export dc/logging-mux -o yaml > $savemuxdc
-fi
-
-function reset_mux_deployconfig() {
-  if [ "$savemuxdc" != "" ]; then
-    os::log::info Restoring original mux deployconfig environment variable
-    os::log::debug "$( oc replace -f $savemuxdc )"
-  fi
+function artifact_log () {
```

Comment:
There is already an `artifact_log` method and an `artifact_out` method e.g. https://github.com/openshift/origin-aggregated-logging/blob/master/hack/testing/util.sh#L295 and https://github.com/openshift/origin-aggregated-logging/blob/master/hack/testing/util.sh#L298

Here is an example: https://github.com/openshift/origin-aggregated-logging/blob/master/test/mux.sh#L51